server.js

```diff
@@ -0,0 +1,291 @@
+    }
+  }
+
+  if (req.method === 'GET') {
+    fs.readFile(`./public${reqURL}`, 'UTF-8', function(err, data) {
+      if (err) throw err;
+      res.writeHead(200, { 'Content-Type': 'text/html', 'Content-Length': data.length });
+      res.write(data);
+      res.end();
+    });
+  }
+
+  if (req.method === 'POST' || req.method === 'PUT' || req.method === 'DELETE') {
+    if (req.headers.authorization === undefined) {
+      res.writeHead(401, { 'WWW-Authenticate': 'Basic realm="Secure Area"' });
+      res.end(`<html><body>Not Authorized</body></html>`);
+      return;
+    } else {
+      let authHeader = req.headers.authorization;
+      let findSpace = authHeader.indexOf(' ');
+      let encodedString = authHeader.slice(findSpace + 1);
+      let base64buffer = new Buffer.from(encodedString, 'base64');
+      let decodedString = base64buffer.toString();
+      let decodedArray = decodedString.split(':');
+
+      if (authenticationObject.hasOwnProperty(decodedArray[0])) {
+        if (authenticationObject[decodedArray[0]] === decodedArray[1]) {
+          console.log('password correct!');
+        } else {
+          res.writeHead(401, { 'WWW-Authenticate': 'Basic realm="Secure Area"' });
+          res.end(`<html><body>Invalid Authentication Credentials</body></html>`);
+          return;
+        }
+      } else {
+        res.writeHead(401, { 'WWW-Authenticate': 'Basic realm="Secure Area"' });
+        res.end(`<html><body>Invalid Authentication Credentials</body></html>`);
+        return;
+      }
+    }
+  }
+
+  if (req.method === 'POST') {
+    if (req.url !== '/elements') {
+      res.writeHead(400, { 'Content-Type': 'text/html' });
+      res.end(`invalid request URL`);
+      return;
+    }
+
+    let body = '';
+    req.on('data', (chunk) => {
+      body += chunk.toString();
+      let bodyObject = qs.decode(body);
+      let bodyKeys = Object.keys(bodyObject);
+
+      if (
+        bodyKeys.indexOf('elementName') !== -1 &&
+        bodyKeys.indexOf('elementSymbol') !== -1 &&
+        bodyKeys.indexOf('elementAtomicNumber') !== -1 &&
+        bodyKeys.indexOf('elementDescription') !== -1
+      ) {
+        elementNameReceived = bodyObject.elementName;
+        elementSymbolReceived = bodyObject.elementSymbol;
+        elementAtomicNumberReceived = bodyObject.elementAtomicNumber;
+        elementDescriptionReceived = bodyObject.elementDescription;
+        fileName = elementNameReceived.toLowerCase();
+
+        let elementTemplate = `<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>The Elements - ${elementNameReceived}</title>
+  <link rel="stylesheet" href="/css/styles.css">
+</head>
+<body>
+  <h1>${elementNameReceived}</h1>
+  <h2>${elementSymbolReceived}</h2>
+  <h3>Atomic number ${elementAtomicNumberReceived}</h3>
+  <p>${elementDescriptionReceived}</p>
+  <p><a href="/">back</a></p>
+</body>`;
+
+        fs.writeFile(`./public/${fileName}.html`, elementTemplate, (err) => {
```

Comment:
From here downward, your fs.readdir should be in the callback of your fs.writeFile.